dune_client/client_async.py

```diff
@@ -372,15 +458,27 @@ async def refresh_into_dataframe(
     async def _get_result_page(
         self,
         job_id: str,
-        limit: int = MAX_NUM_ROWS_PER_BATCH,
-        offset: int = 0,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+        columns: Optional[List[str]] = None,
+        sample_count: Optional[int] = None,
+        filters: Optional[str] = None,
+        sort_by: Optional[List[str]] = None,
     ) -> ResultsResponse:
         """GET a page of results from Dune API for `job_id` (aka `execution_id`)"""
 
-        params = {
-            "limit": limit,
-            "offset": offset,
-        }
+        if sample_count is None and limit is None and offset is None:
```

Comment:
I've seen this if or similar a few times, can it be part of `_build_parameters()` instead ?